src/app/auth/permission.service.ts

```diff
@@ -47,7 +47,7 @@ export class PermissionService {
    */
   hasScope(resourceId: string, scope: string): Observable<boolean> {
     return this.getPermission(resourceId).pipe(
-      map((permission: Permission) => permission.scopes.includes(scope))
+      map((permission: Permission | undefined) => permission.scopes.includes(scope))
```

Comment:
This is still a problem because you use `permission` without a null check. The old code use to return `false` if `permission` was `null`.